scipy/_lib/_array_api_compat_vendor.py

```diff
@@ -0,0 +1,11 @@
+# DO NOT RENAME THIS FILE
+# This is a hook for array_api_extra/src/array_api_extra/_lib/_compat.py
+# to override functions of array_api_compat.
+
+from __future__ import annotations
```

Comment:
is this line required? https://github.com/scipy/scipy/pull/21981 purged this from SciPy, and https://github.com/data-apis/array-api-extra/pull/61 did the same for array-api-extra. Unless it is required for an interaction with array-api-compat?